File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
input = sys.stdin.readline

def dfs(start):
    visited[start] = True
    print(start, end=" ")

    for i in range(1, node+1):
        if(graph[start][i] == True and visited[i] == False):    # 아래 노드가 연결되어 있고, 방문하지 않은 경우
            dfs(i)


def bfs(start):
    que = deque([start])
    visited[start] = True

    while que:
        v = que.popleft()
        print(v, end=" ")
        
        for i in range(1, node+1):
            if(graph[v][i] == True and visited[i] == False):    # 다음 노드와 연결되어 있고, 방문하지 않았을 때
                que.append(i)
                visited[i] = True   # 노드 방문



node, edge, num = map(int, input().split())
graph = [[False]*(node+1) for _ in range(node+1)]   # 그래프 행렬 생성

for i in range(edge):
    n1, n2 = map(int, input().split())
    graph[n1][n2] = graph[n2][n1] = True    # 연결되어 있음 
    

visited = [False]*(node+1)  # 방문 확인 list
dfs(num)
print() 

visited = [False]*(node+1) 
bfs(num)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static ArrayList<Integer>[] graph;
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int v = Integer.parseInt(st.nextToken());

        graph = new ArrayList[n+1];

        for(int i=1; i<=n; i++){
            graph[i] = new ArrayList<>();    // 초기화
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int node1 = Integer.parseInt(st.nextToken());
            int node2 = Integer.parseInt(st.nextToken());
            graph[node1].add(node2);
            graph[node2].add(node1);
        }

        for(int i=1; i<=n; i++){
            Collections.sort(graph[i]);
        }

        visited = new boolean[n+1];
        dfs(v);
        System.out.println();
        visited = new boolean[n+1];
        bfs(v);
    }

    public static void dfs(int v){
        System.out.print(v + " ");
        visited[v] = true;

        for(int w : graph[v]){
            if(!visited[w]){
                dfs(w);
            }
        }
    }

    public static void bfs(int v){
        Queue<Integer> q = new LinkedList<>();
        q.add(v);
        visited[v] = true;

        while(!q.isEmpty()){
            int w = q.poll();
            System.out.print(w + " ");
            for(int x : graph[w]){
                if(!visited[x]){
                    q.add(x);
                    visited[x] = true;
                }
            }
        }
    }
}

